src/Microsoft.Kusto.ServiceLayer/DataSource/DataSourceFactory.cs

```diff
@@ -26,7 +26,8 @@ public IDataSource Create(DataSourceType dataSourceType, ConnectionDetails conne
                 {
                     var kustoConnectionDetails = MapKustoConnectionDetails(connectionDetails);
                     var kustoClient = new KustoClient(kustoConnectionDetails, ownerUri);
-                    return new KustoDataSource(kustoClient);
+                    var intellisenseClient = new KustoIntellisenseClient(kustoClient);
+                    return new KustoDataSource(kustoClient, intellisenseClient);
```

Comment:
Separated database interaction and syntax highlighting with intellisense. Both will still live on the KustoDataSource.